packages/engine-core/src/data-proxy/utils/getClientVersion.ts

```diff
@@ -1,22 +1,34 @@
 import type { EngineConfig } from '../../common/Engine'
+import { NotImplementedYetError } from '../errors/NotImplementedYetError'
 
 /**
  * Determine the client version to be sent to the DataProxy
  * @param config
  * @returns
  */
-export function getClientVersion(config: EngineConfig) {
-  const [version, suffix] = config.clientVersion?.split('-') ?? []
-
-  // we expect the version to match the pattern major.minor.patch
-  if (!suffix && /^[1-9][0-9]*\.[0-9]+\.[0-9]+$/.test(version)) {
-    return version
+export function getClientVersion(config: EngineConfig): string {
+  if (process.env.PRISMA_CLIENT_DATA_PROXY_CLIENT_VERSION) {
+    // should we log with debug here that this env var is used?
+    return process.env.PRISMA_CLIENT_DATA_PROXY_CLIENT_VERSION
+  } else if (config.clientVersion) {
+    const [version, suffix] = config.clientVersion.split('-') ?? []
+    // we expect the version to match the pattern major.minor.patch
+    if (!suffix && /^[1-9][0-9]*\.[0-9]+\.[0-9]+$/.test(version)) {
+      return version
+    } else {
+      throw new NotImplementedYetError('Only `major.minor.patch` versions are supported.', {
+        clientVersion: config.clientVersion,
+      })
+    }
```

Comment:
These checks should probably be applied to `process.env.PRISMA_CLIENT_DATA_PROXY_CLIENT_VERSION` as well.

Reply:
As the env var is for internal use only really, not doing that would open up us being able to use _any_ type of version identifier the DPD might or might not support.